collector/database/chart.php

```diff
@@ -59,16 +65,57 @@
     }
 }
 
-$select = new \single_select(
+$select = new single_select(
     $url,
     'metric',
     $metriclabels,
     $metricname
 );
 $select->set_label(get_string('select_metric_for_display', 'cltr_database'));
+$context = [];
+// Error management if metric is not enabled.
+if (!isset($metrics[$metricname])) {
+    throw new moodle_exception('metric_not_enabled', 'tool_cloudmetrics', '', $metricname);
+}
+$backfilledmetric = $metrics[$metricname]->is_backfillable();
```

Comment:
This page is about showing the chart, I'd prefer it if all the logic around backfilling was moved to a new endpoint which you post to, and then after it has filled the data then it redirects back to the chart page as a get.

Reply:
I have split those in two pages chart + backfill